sale_recurring_payment/models/sale_subscription.py

```diff
@@ -109,19 +107,19 @@ def write(self, values):
                 if (
                     record.stage_id
                     and record.stage_id.type == "post"
-                    and record.payment_provider_subscription_id
-                    and record.is_payment_provider_subscription_terminated
+                    and record.payment_provider_mandate_id
+                    and record.is_payment_provider_mandate_terminated
                 ):
                     raise UserError(
                         _(
-                            "Terminated subscriptions with payment provider subscription also terminated cannot be "
+                            "Terminated subscriptions with payment provider mandate also terminated cannot be "
```

Comment:
A UX consideration is to recreate the mandate here if this is the case instead of throwing an user error. Non-blocking issue